--- a/bootstrap
+++ b/bootstrap
@@ -15,20 +15,3 @@
     ${AUTORECONF} --verbose --install --symlink \
         || exit $?
 fi
-
-# Configure OpenPACE
-if ! test -r Makefile
-then
-	./configure \
-        || exit $?
-fi
-
-# Build OpenPACE
-if ! test -x src/eactest
-then
-    ${MAKE} \
-        || exit $?
-fi
-
-src/eactest \
-    || exit $?
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -1,13 +1,13 @@
 SUBDIRS = docs certs
 
-BUILT_SOURCES = cvc-create-cmdline.c cvc-create-cmdline.h cvc-print-cmdline.c cvc-print-cmdline.h
+BUILT_SOURCES =
 
 EXTRA_DIST = bsi_objects.txt cvc-print.ggo.in cvc-create.ggo.in cvc-create.ggo cvc-print.ggo $(BUILT_SOURCES)
 
 do_subst = $(SED) \
 	   -e 's,[@]PACKAGE_BUGREPORT[@],$(PACKAGE_BUGREPORT),g'
 
-dist_man1_MANS		= cvc-create.1 cvc-print.1
+dist_man1_MANS		=
 
 MAINTAINERCLEANFILES = $(BUILT_SOURCES) cvc-create.ggo cvc-print.ggo $(dist_man1_MANS)
 
@@ -43,45 +43,13 @@
 libeac_la_LIBADD   = $(CRYPTO_LIBS)
 if WIN32
 AM_CPPFLAGS        = -DWIN32_LEAN_AND_MEAN
-libeac_la_LIBADD  += -lws2_32
+libeac_la_LIBADD  += -leOI32.dll -lws2_32
 endif
 
 libvc_la_SOURCES  = vc.c
 
 
-bin_PROGRAMS         = eactest cvc-print cvc-create example
-
-EAC_LIBS             = libeac.la libvc.la $(CRYPTO_LIBS)
-EAC_CFLAGS           = $(CRYPTO_CFLAGS)
-
-eactest_SOURCES      = eactest.c ssl_compat.c
-eactest_LDADD        = $(EAC_LIBS)
-eactest_CFLAGS       = $(EAC_CFLAGS)
-
-cvc_print_SOURCES     = cvc-print.c read_file.c cvc-print-cmdline.c
-cvc_print_LDADD       = $(EAC_LIBS)
-cvc_print_CFLAGS      = $(EAC_CFLAGS)
-
-cvc_create_SOURCES     = cvc-create.c read_file.c cvc-create-cmdline.c
-cvc_create_LDADD       = $(EAC_LIBS)
-cvc_create_CFLAGS      = $(EAC_CFLAGS)
-
-example_SOURCES      = example.c
-example_LDADD        = $(EAC_LIBS)
-example_CFLAGS       = $(EAC_CFLAGS)
-
-
-cvc-create-cmdline.c cvc-create-cmdline.h: cvc-create.ggo
-	$(GENGETOPT) --file-name=cvc-create-cmdline --output-dir=$(builddir) < $<
-
-cvc-print-cmdline.c cvc-print-cmdline.h: cvc-print.ggo
-	$(GENGETOPT) --file-name=cvc-print-cmdline --output-dir=$(builddir) < $<
-
-cvc-create.ggo: cvc-create.ggo.in
-	@$(do_subst) < $< > $(builddir)/$@
-
-cvc-print.ggo: cvc-print.ggo.in
-	@$(do_subst) < $< > $(builddir)/$@
+bin_PROGRAMS         =
 
 
 
