--- a/configure.ac
+++ b/configure.ac
@@ -775,13 +775,13 @@
 LIBS="$LDFLAGS $OPENPACE_LIBS"
 
 have_openpace="yes"
-AC_CHECK_HEADERS(eac/eac.h, [],
-                 [ AC_MSG_WARN([OpenPACE headers not found])
-                 have_openpace="no" ])
-AC_MSG_CHECKING([for EAC_CTX_init_pace])
-AC_TRY_LINK_FUNC(EAC_CTX_init_pace, [ AC_MSG_RESULT([yes]) ],
-                 [ AC_MSG_WARN([Cannot link against libeac])
-                 have_openpace="no" ])
+#AC_CHECK_HEADERS(eac/eac.h, [],
+#                 [ AC_MSG_WARN([OpenPACE headers not found])
+#                 have_openpace="no" ])
+#AC_MSG_CHECKING([for EAC_CTX_init_pace])
+#AC_TRY_LINK_FUNC(EAC_CTX_init_pace, [ AC_MSG_RESULT([yes]) ],
+#                 [ AC_MSG_WARN([Cannot link against libeac])
+#                 have_openpace="no" ])
 AC_CHECK_FUNCS([EAC_OBJ_nid2obj])
 
 CPPFLAGS="$saved_CPPFLAGS"
--- a/src/minidriver/Makefile.am
+++ b/src/minidriver/Makefile.am
@@ -12,13 +12,13 @@
 dist_noinst_DATA = eOI-minidriver.inf
 endif
 
-AM_CFLAGS = $(OPTIONAL_OPENSSL_CFLAGS)
+AM_CFLAGS = $(OPTIONAL_OPENSSL_CFLAGS) $(OPENPACE_CFLAGS)
 AM_CPPFLAGS = -I$(top_srcdir)/src
 
 eOI_minidriver@LIBRARY_BITNESS@_la_SOURCES = minidriver.c minidriver.exports versioninfo-minidriver.rc
 eOI_minidriver@LIBRARY_BITNESS@_la_LIBADD =  \
 	$(top_builddir)/src/libopensc/libeOI_static.la \
-	$(OPTIONAL_OPENSSL_LIBS) \
+	$(OPTIONAL_OPENSSL_LIBS) $(OPENPACE_LIBS) \
 	-lbcrypt -lcrypt32 -lrpcrt4
 eOI_minidriver@LIBRARY_BITNESS@_la_LDFLAGS = $(AM_LDFLAGS) \
 	-export-symbols "$(srcdir)/minidriver.exports" \
--- a/src/tools/Makefile.am
+++ b/src/tools/Makefile.am
@@ -56,8 +56,8 @@
 	$(top_builddir)/src/common/libcompat.la
 
 sceac_example_SOURCES = sceac-example.c
-sceac_example_LDADD = $(top_builddir)/src/libopensc/libeOI.la $(OPENPACE_LIBS)
-sceac_example_CFLAGS = -I$(top_srcdir)/src $(OPENPACE_CFLAGS)
+sceac_example_LDADD = $(top_builddir)/src/libopensc/libeOI.la $(OPENPACE_LIBS) $(OPENSSL_LIBS)
+sceac_example_CFLAGS = -I$(top_srcdir)/src $(OPENPACE_CFLAGS) $(OPENSSL_CFLAGS)
 
 eOI_tool_SOURCES = opensc-tool.c util.c
 piv_tool_SOURCES = piv-tool.c util.c
@@ -101,8 +101,8 @@
 gids_tool_LDADD = $(OPTIONAL_OPENSSL_LIBS)
 
 npa_tool_SOURCES = npa-tool.c fread_to_eof.c util.c npa-tool-cmdline.c
-npa_tool_LDADD = $(OPENPACE_LIBS)
-npa_tool_CFLAGS = $(OPENPACE_CFLAGS)
+npa_tool_LDADD = $(OPENPACE_LIBS) $(OPENSSL_LIBS)
+npa_tool_CFLAGS = $(OPENPACE_CFLAGS) $(OPENSSL_CFLAGS)
 if HAVE_UNKNOWN_WARNING_OPTION
 npa_tool_CFLAGS += -Wno-unknown-warning-option
 endif
