--- a/src/libopensc/card-eoi.c
+++ b/src/libopensc/card-eoi.c
@@ -45,6 +45,8 @@
 	{384, {{1, 3, 132, 0, 34, -1}}}
 };
 
+static char *eoi_model = "ChipDocLite";
+
 /* The description of the driver. */
 static struct sc_card_driver eoi_drv =
 {
@@ -415,6 +417,27 @@
 	LOG_FUNC_RETURN(card->ctx, sc_get_iso7816_driver()->ops->pin_cmd(card, data, tries_left));
 }
 
+static int eoi_card_ctl(sc_card_t *card, unsigned long cmd, void *ptr)
+{
+	int r = 0;
+
+	LOG_FUNC_CALLED(card->ctx);
+	switch (cmd)
+	{
+	case SC_CARDCTL_GET_MODEL:
+		if (!ptr)
+			r = SC_ERROR_INVALID_ARGUMENTS;
+		else {
+			*(char**)ptr = eoi_model;
+			r = SC_SUCCESS;
+		}
+		break;
+	default:
+		r = sc_get_iso7816_driver()->ops->card_ctl(card, cmd, ptr);
+	}
+	LOG_FUNC_RETURN(card->ctx, r);
+}
+
 #define ALREADY_PROCESSED 0x80000000
 
 static int eoi_set_security_env(struct sc_card *card, const struct sc_security_env *env, int se_num)
@@ -543,6 +566,7 @@
 	eoi_ops.select_file = eoi_select_file;
 	eoi_ops.logout = eoi_logout;
 	eoi_ops.pin_cmd = eoi_pin_cmd;
+	eoi_ops.card_ctl = eoi_card_ctl;
 	eoi_ops.set_security_env = eoi_set_security_env;
 	eoi_ops.compute_signature = eoi_compute_signature;
 
diff --git a/src/libopensc/cardctl.h b/src/libopensc/cardctl.h
index 32b4c68f..fe1c18e1 100644
--- a/src/libopensc/cardctl.h
+++ b/src/libopensc/cardctl.h
@@ -46,6 +46,7 @@
 	SC_CARDCTL_GET_CHV_REFERENCE_IN_SE,
 	SC_CARDCTL_PKCS11_INIT_TOKEN,
 	SC_CARDCTL_PKCS11_INIT_PIN,
+	SC_CARDCTL_GET_MODEL,
 
 	/*
 	 * GPK specific calls
diff --git a/src/pkcs11/framework-pkcs15.c b/src/pkcs11/framework-pkcs15.c
index af6836b0..656bfadc 100644
--- a/src/pkcs11/framework-pkcs15.c
+++ b/src/pkcs11/framework-pkcs15.c
@@ -415,6 +415,8 @@
 
 	if (model)
 		strcpy_bp(pToken->model, model, sizeof(pToken->model));
+	else if (sc_card_ctl(p15card->card, SC_CARDCTL_GET_MODEL, &model) == SC_SUCCESS)
+		strcpy_bp(pToken->model, model, sizeof(pToken->model));
 	else if (p15card->flags & SC_PKCS15_CARD_FLAG_EMULATED)
 		strcpy_bp(pToken->model, "PKCS#15 emulated", sizeof(pToken->model));
 	else

